### PR TITLE
feat: update the config to use Hikari for connection pool configuration per use case

### PR DESCRIPTION
**Pull request by tisla1080**
Replace the `DriverManagerDataSource` with `HikariDataSource` to allow users to customize connection pool settings as needed by overriding the `addAdditionalProperties()` method

- use HikariDataSource
- update pom file with version properties
- exclude tomcat jdbc from jpa dependency
- include synapse utilities common
- add sample service that uses the synapse data db2 sample

### CODE REVIEW

**Review by eaquin02 — APPROVED** [review]

File: data/synapse-data-db2/src/main/java/io/americanexpress/synapse/data/db2/entity/BaseDb2Entity.java [hunk 1]
```diff
@@ -23,25 +22,4 @@
 @MappedSuperclass
 public abstract class BaseDb2Entity {
 
-    /**
-     * The id of the entity.
-     */
-    @Id
-    private Long id;
```
Comment (jmartinez5120) [inline data/synapse-data-db2/src/main/java/io/americanexpress/synapse/data/db2/entity/BaseDb2Entity.java]: Synapse architecture shouldn't change for a use case or a POD system. That would compromise other apps architecture and design. I would recommend an alternative to change synapse.

**Review by jmartinez5120 — CHANGES_REQUESTED** [review]
Please look at my comments.

File: data/synapse-data-db2/src/main/java/io/americanexpress/synapse/data/db2/entity/BaseDb2Entity.java (hunk 1 above)
Comment (tisla1080) [reply data/synapse-data-db2/src/main/java/io/americanexpress/synapse/data/db2/entity/BaseDb2Entity.java]: reverted

File: service/service-samples/sample-service-db2-book/pom.xml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            <groupId>io.americanexpress.synapse</groupId>
+            <artifactId>sample-data-db2-book</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.americanexpress.synapse</groupId>
+            <artifactId>synapse-service-rest</artifactId>
+        </dependency>
+    </dependencies>
+
+</project>
```
Comment (guttff) [inline service/service-samples/sample-service-db2-book/pom.xml]: add an extra line at end of file

File: service/service-samples/sample-service-db2-book/src/test/resources/Sample DB2 Service Synapse.postman_collection.json [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+					"path": [
+						"v1",
+						"books"
+					]
+				}
+			},
+			"response": []
+		}
+	]
+}
```
Comment (guttff) [inline service/service-samples/sample-service-db2-book/src/test/resources/Sample DB2 Service Synapse.postman_collection.json]: add an extra line at end of file

File: data/synapse-data-db2/pom.xml [hunk 4]
```diff
@@ -23,10 +23,27 @@
...
+                <exclusion>
+                    <groupId>org.apache.tomcat</groupId>
+                    <artifactId>tomcat-jdbc</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>io.americanexpress.synapse</groupId>
+            <artifactId>synapse-utilities-common</artifactId>
+            <version>${synapse.version}</version>
```
Comment (guttff) [inline data/synapse-data-db2/pom.xml]: remove version, this should already be specify in the parent pom

File: data/synapse-data-db2/pom.xml [hunk 5]
```diff
@@ -35,12 +52,12 @@
         <dependency>
             <groupId>org.hibernate</groupId>
             <artifactId>hibernate-core</artifactId>
-            <version>6.1.7.Final</version>
+            <version>${hibernate.core.version}</version>
```
Comment (guttff) [inline data/synapse-data-db2/pom.xml]: emove version, this should already be specify in the parent pom

File: data/synapse-data-db2/pom.xml [hunk 6]
```diff
@@ -35,12 +52,12 @@
...
         </dependency>
         <dependency>
             <groupId>org.hibernate</groupId>
             <artifactId>hibernate-entitymanager</artifactId>
-            <version>5.6.15.Final</version>
+            <version>${hibernate.entitymanager.version}</version>
```
Comment (guttff) [inline data/synapse-data-db2/pom.xml]: emove version, this should already be specify in the parent pom

File: data/synapse-data-db2/pom.xml [hunk 7]
```diff
@@ -23,10 +23,27 @@
     <modelVersion>4.0.0</modelVersion>
     <artifactId>synapse-data-db2</artifactId>
 
+    <properties>
+        <synapse.version>0.3.30-SNAPSHOT</synapse.version>
+        <hibernate.core.version>5.6.15.Final</hibernate.core.version>
+        <hibernate.entitymanager.version>5.6.15.Final</hibernate.entitymanager.version>
```
Comment (guttff) [inline data/synapse-data-db2/pom.xml]: remove version, this should already be specify in the parent pom

File: service/service-samples/sample-service-db2-book/src/main/java/io/americanexpress/service/book/rest/config/BookConfig.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+import org.springframework.context.annotation.PropertySource;
+
+/**
+ * {@code BookConfig} sets configuration and enables database querying.
+ */
+@ComponentScan(basePackages = "io.americanexpress.service.book.rest")
+@Configuration
+@Import({BookDataConfig.class, ServiceRestConfig.class})
+@PropertySource("classpath:/service-book-application.properties")
+public class BookConfig {
```
Comment (guttff) [inline service/service-samples/sample-service-db2-book/src/main/java/io/americanexpress/service/book/rest/config/BookConfig.java]: where is the base service config extention

File: service/service-samples/sample-service-db2-book/src/main/java/io/americanexpress/service/book/rest/model/BookRequest.java [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+ * the License.
+ */
+package io.americanexpress.service.book.rest.model;
+
+import io.americanexpress.synapse.service.rest.model.BaseServiceRequest;
+
+/**
+ * {@code BookRequest} is the base book request pojo.
+ */
+public class BookRequest implements BaseServiceRequest {
```
Comment (guttff) [inline service/service-samples/sample-service-db2-book/src/main/java/io/americanexpress/service/book/rest/model/BookRequest.java]:
```suggestion
public class BookServiceRequest implements BaseServiceRequest {
```

File: service/service-samples/sample-service-db2-book/src/main/java/io/americanexpress/service/book/rest/model/BookResponse.java [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+ * the License.
+ */
+package io.americanexpress.service.book.rest.model;
+
+import io.americanexpress.synapse.service.rest.model.BaseServiceResponse;
+
+/**
+ * {@code BookResponse} class is the response object for the book service.
+ */
+public class BookResponse extends BaseServiceResponse {
```
Comment (guttff) [inline service/service-samples/sample-service-db2-book/src/main/java/io/americanexpress/service/book/rest/model/BookResponse.java]:
```suggestion
public class BookServiceResponse extends BaseServiceResponse {
```

**Review by guttff — CHANGES_REQUESTED** [review]

File: service/service-samples/sample-service-db2-book/pom.xml (hunk 2 above)
Comment (tisla1080) [reply service/service-samples/sample-service-db2-book/pom.xml]: done

File: service/service-samples/sample-service-db2-book/src/test/resources/Sample DB2 Service Synapse.postman_collection.json (hunk 3 above)
Comment (tisla1080) [reply service/service-samples/sample-service-db2-book/src/test/resources/Sample DB2 Service Synapse.postman_collection.json]: done

File: service/service-samples/sample-service-db2-book/src/main/java/io/americanexpress/service/book/rest/config/BookConfig.java (hunk 8 above)
Comment (tisla1080) [reply service/service-samples/sample-service-db2-book/src/main/java/io/americanexpress/service/book/rest/config/BookConfig.java]: extended and fixed the @Import

File: service/service-samples/sample-service-db2-book/src/main/java/io/americanexpress/service/book/rest/model/BookResponse.java (hunk 10 above)
Comment (tisla1080) [reply service/service-samples/sample-service-db2-book/src/main/java/io/americanexpress/service/book/rest/model/BookResponse.java]: done

File: service/service-samples/sample-service-db2-book/src/main/java/io/americanexpress/service/book/rest/model/BookRequest.java (hunk 9 above)
Comment (tisla1080) [reply service/service-samples/sample-service-db2-book/src/main/java/io/americanexpress/service/book/rest/model/BookRequest.java]: done

File: data/synapse-data-db2/pom.xml (hunk 5 above)
Comment (tisla1080) [reply data/synapse-data-db2/pom.xml]: done

File: data/synapse-data-db2/pom.xml (hunk 6 above)
Comment (tisla1080) [reply data/synapse-data-db2/pom.xml]: done

File: data/synapse-data-db2/pom.xml (hunk 7 above)
Comment (tisla1080) [reply data/synapse-data-db2/pom.xml]: done

File: data/synapse-data-db2/pom.xml (hunk 4 above)
Comment (tisla1080) [reply data/synapse-data-db2/pom.xml]: done

**Review by guttff — APPROVED** [review]

**Pull request merged**